temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -697,8 +697,8 @@ public boolean start() {
 
   @Override
   public CompletableFuture<Void> shutdown(ShutdownManager shutdownManager, boolean interruptTasks) {
+    slotQueue.shutdown();
```

Comment:
I think the error here is the `slotQueue` is starting a thread in the constructor unconditionally. The `slotQueue` should only be started in the workers `start` if the local activity worker has any handles. 

cc @Sushisource

Reply:
Pay attention that `awaitTermination` call this unconditionally.

Reply:
I mention your suggestion in original issue as one of the resoutions.

But I more worry about other usages of `slotQueue` (`submitAttempt`, `submitANewExecution`). 

For me it's not so easy to make sure that they can't be called without `start`, because if they can we will introduce NPE.

Reply:
`submitAttempt ` and `submitANewExecution ` cannot be called without start. Also `slotQueue` should still be created in the constructor, I am just saying the thread it creates should not be started until the worker is started.

Reply:
Then, I think we can introduce multistep initialization here.
* constructor will initialize object, without starting a thread
* introduce start method which will start listen thread and will be called from worker start


WDYT?

Reply:
Yes exactly

Reply:
I'll add a test to make sure whatever we end up with has the fix. Thanks for your effort on this @nikoncode !